lib/ace/ext/papeeria/highlighter.coffee

```diff
@@ -76,6 +77,20 @@ define((require, exports, module) ->
                     else
                         rightNearest = rightCandidate
             key++
+
+        #Special case related to positionLeftwards has not next char and positionLeftWards.column += 1 does nothing.
+        if session.getLine(positionRightwards.row + 1) == '' 
```

Comment:
First, `rightNearest == null` is also the case when the text after cursor is properly balanced.
Second, I still don't understand why do we deal with the right bracket here if the issue was with the left one (and comment in the line above also says that this is the case when hach with changing positionLeftwards doesn't work).